### PR TITLE
At least & at most

### PR DESCRIPTION
**Pull request by xapn**
I added two methods about expectations on the number of method invocations.
- Expect the last invocation at least a minimal number of times.
- Expect the last invocation at most a maximal number of times.

### CODE REVIEW

**Comment by jolkdarr** [comment]
Java consistency: all "Expect" should be replaced by "Expects" (see [Javadoc guidelines](http://www.oracle.com/technetwork/articles/java/index-137868.html)).

**Comment by xapn** [comment]
I fix up it.

File: core/src/main/java/org/easymock/internal/MocksControl.java [hunk 1]
```diff
@@ -373,6 +373,24 @@ public void asStub() {
...
+            state.times(new Range(minimalNumberOfTimes, Integer.MAX_VALUE));
+            return this;
+        } catch (RuntimeExceptionWrapper e) {
+            throw (RuntimeException) e.getRuntimeException().fillInStackTrace();
+        }
+    }
+
+    public IExpectationSetters<Object> atMost(int maximalNumberOfTimes) {
+        try {
+            state.times(new Range(1, maximalNumberOfTimes));
```
Comment (BlueIce) [inline core/src/main/java/org/easymock/internal/MocksControl.java]: Should the range start at zero or at one? From the method name i would have expected that never calling the mock is included in this expectation.

Comment (xapn) [inline core/src/main/java/org/easymock/internal/MocksControl.java]: "atMost" should make sens only from 1. Otherwise, it would be "never".

Comment (BlueIce) [inline core/src/main/java/org/easymock/internal/MocksControl.java]: Let me rephrase this:
if i do `atMost(5)` but i never call the mock then the test will fail, right?
My argument is that the test should not fail and hence line 387 should read
```
state.times(new Range(0, maximalNumberOfTimes));
```

Comment (xapn) [inline core/src/main/java/org/easymock/internal/MocksControl.java]: Ok, I understand.
I remember that the Range class did not support a minimum lesser than 1. Maybe it is related to the fact mocks are typed by EasyMock.

Comment (henri-tremblay) [inline core/src/main/java/org/easymock/internal/MocksControl.java]: The minimum range is 0. However, the maximum can't be 0 for some historical
reason. So you can't have a `times(0, 0)` range (which would mean: "I want
that to be not called")
